### PR TITLE
r/iam_server_certificate - tagging support

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #17594

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSIAMServerCertificate_'
--- PASS: TestAccAWSIAMServerCertificate_disappears (29.49s)
--- PASS: TestAccAWSIAMServerCertificate_name_prefix (34.99s)
--- PASS: TestAccAWSIAMServerCertificate_basic (39.60s)
--- PASS: TestAccAWSIAMServerCertificate_Path (39.84s)
--- PASS: TestAccAWSIAMServerCertificate_file (60.58s)
```

### CODE REVIEW

File: website/docs/r/iam_server_certificate.html.markdown [hunk 1]
```diff
@@ -117,6 +117,7 @@ In addition to all arguments above, the following attributes are exported:
 * `id` - The unique Server Certificate name
 * `name` - The name of the Server Certificate
 * `arn` - The Amazon Resource Name (ARN) specifying the server certificate.
+* `tags` - Key-value mapping of tags for the server certificate.
```
Comment (ewbankkit) [inline website/docs/r/iam_server_certificate.html.markdown:120]: Add documentation for new `expiration` and `upload_date` attributes.

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSIAMServerCertificate_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSIAMServerCertificate_ -timeout 120m
=== RUN   TestAccAWSIAMServerCertificate_basic
=== PAUSE TestAccAWSIAMServerCertificate_basic
=== RUN   TestAccAWSIAMServerCertificate_tags
=== PAUSE TestAccAWSIAMServerCertificate_tags
=== RUN   TestAccAWSIAMServerCertificate_name_prefix
=== PAUSE TestAccAWSIAMServerCertificate_name_prefix
=== RUN   TestAccAWSIAMServerCertificate_disappears
=== PAUSE TestAccAWSIAMServerCertificate_disappears
=== RUN   TestAccAWSIAMServerCertificate_file
=== PAUSE TestAccAWSIAMServerCertificate_file
=== RUN   TestAccAWSIAMServerCertificate_Path
=== PAUSE TestAccAWSIAMServerCertificate_Path
=== CONT  TestAccAWSIAMServerCertificate_basic
=== CONT  TestAccAWSIAMServerCertificate_Path
=== CONT  TestAccAWSIAMServerCertificate_disappears
=== CONT  TestAccAWSIAMServerCertificate_file
=== CONT  TestAccAWSIAMServerCertificate_name_prefix
=== CONT  TestAccAWSIAMServerCertificate_tags
--- PASS: TestAccAWSIAMServerCertificate_disappears (20.31s)
--- PASS: TestAccAWSIAMServerCertificate_name_prefix (22.18s)
--- PASS: TestAccAWSIAMServerCertificate_basic (24.44s)
--- PASS: TestAccAWSIAMServerCertificate_Path (24.61s)
--- PASS: TestAccAWSIAMServerCertificate_file (32.74s)
--- PASS: TestAccAWSIAMServerCertificate_tags (40.73s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	41.666s
```

File: website/docs/r/iam_server_certificate.html.markdown (hunk 1 above)
Comment (DrFaust92) [reply website/docs/r/iam_server_certificate.html.markdown:120]: Added

File: aws/resource_aws_iam_server_certificate.go [hunk 2]
```diff
@@ -140,16 +153,38 @@ func resourceAwsIAMServerCertificateRead(d *schema.ResourceData, meta interface{
...
-	d.Set("certificate_body", resp.ServerCertificate.CertificateBody)
-	d.Set("certificate_chain", resp.ServerCertificate.CertificateChain)
-	d.Set("path", resp.ServerCertificate.ServerCertificateMetadata.Path)
-	d.Set("arn", resp.ServerCertificate.ServerCertificateMetadata.Arn)
+	d.Set("certificate_body", cert.CertificateBody)
+	d.Set("certificate_chain", cert.CertificateChain)
+	d.Set("path", metadata.Path)
+	d.Set("arn", metadata.Arn)
+	d.Set("expiration", aws.TimeValue(metadata.Expiration).Format(time.RFC3339))
+	d.Set("upload_date", aws.TimeValue(metadata.UploadDate).Format(time.RFC3339))
```
Comment (bflad) [inline aws/resource_aws_iam_server_certificate.go:165]: Nit: Time values may be `nil`, which will return a confusing timestamp from formatting a "0" time. Reference: https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#root-typestring-and-aws-timestamp

```suggestion
	if metadata.Expiration != nil {
		d.Set("expiration", aws.TimeValue(metadata.Expiration).Format(time.RFC3339))
	} else {
		d.Set("expiration", nil)
	}

	if metadata.UploadDate != nil {
		d.Set("upload_date", aws.TimeValue(metadata.UploadDate).Format(time.RFC3339))
	} else {
		d.Set("upload_date", nil)
	}
```

File: website/docs/r/iam_server_certificate.html.markdown (hunk 1 above)
Comment (bflad) [reply website/docs/r/iam_server_certificate.html.markdown:120]: Tags should be documented in the arguments section -- will move up on merge.

**Review by bflad — APPROVED** [review]
Thank you, @DrFaust92! 🚀 

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSIAMServerCertificate_disappears (14.13s)
--- PASS: TestAccAWSIAMServerCertificate_name_prefix (15.89s)
--- PASS: TestAccAWSIAMServerCertificate_Path (18.27s)
--- PASS: TestAccAWSIAMServerCertificate_basic (18.19s)
--- PASS: TestAccAWSIAMServerCertificate_file (31.71s)
--- PASS: TestAccAWSIAMServerCertificate_tags (40.61s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSIAMServerCertificate_disappears (17.41s)
--- PASS: TestAccAWSIAMServerCertificate_name_prefix (20.05s)
--- PASS: TestAccAWSIAMServerCertificate_basic (22.42s)
--- PASS: TestAccAWSIAMServerCertificate_Path (22.36s)
--- PASS: TestAccAWSIAMServerCertificate_file (37.52s)
--- PASS: TestAccAWSIAMServerCertificate_tags (51.54s)
```

**Pull request merged**